tests/evm/test_create.py

```diff
@@ -0,0 +1,340 @@
+            .create(
+                stack.value,
+                stack.offset,
+                stack.size,
+            )
+            .stop()
+        )
+
+    callee_bytecode = callee_init_codes
+    callee_bytecode_hash = RLC(callee_bytecode.hash(), randomness)
+
+    nonce = caller.nonce
+    is_success = True if callee_init_codes is RETURN_BYTECODE else True
+    is_reverted_by_caller = not caller_ctx.is_persistent and is_success
+    is_reverted_by_callee = not is_success
+    callee_is_persistent = caller_ctx.is_persistent and is_success
+    callee_rw_counter_end_of_reversion = (
+        80
+        if is_reverted_by_callee
+        else (
+            caller_ctx.rw_counter_end_of_reversion - (caller_ctx.reversible_write_counter + 1)
+            if is_reverted_by_caller
+            else 0
+        )
+    )
+
+    contract_addr = keccak256(rlp.encode([caller.address.to_bytes(20, "big"), caller.nonce]))
+    contract_address = int.from_bytes(contract_addr[-20:], "big")
+
+    # can't be a static all
+    is_static = 0
+
+    call_id = 65
+    rw_counter = call_id
+    # CREATE: 33 * (3(push) + 2(push from mstore)) + 1(CREATE) + 1(mstore)
+    # CREATE2: 33 * (4(push) + 2(push from mstore)) + 1(CREATE) + 1(mstore)
+    next_program_counter = 33 * 6 + 1 + 1 if is_create2 else 33 * 5 + 1 + 1
+    assert caller_bytecode.code[next_program_counter - 1] == Opcode.CREATE
+    # CREATE: 1024 - 3 + 1 = 1022
+    # CREATE2: 1024 - 4 + 1 = 1021
+    stack_pointer = 1021 - is_create2
+
+    rw_dictionary = RWDictionary(rw_counter)
+    # fmt: off
+    # stack
+    if is_create2:
+        rw_dictionary.stack_read(1, 1020, RLC(stack.salt, randomness))
+    rw_dictionary \
+        .stack_read(1, 1021, RLC(stack.value, randomness)) \
+        .stack_read(1, 1022, RLC(stack.offset, randomness)) \
+        .stack_read(1, 1023, RLC(stack.size, randomness)) \
+        .stack_write(1, 1023, RLC(contract_address, randomness)) \
+
+    # caller's call context
+    rw_dictionary \
+        .call_context_read(1, CallContextFieldTag.Depth, 1) \
+        .call_context_read(1, CallContextFieldTag.TxId, 1) \
+        .call_context_read(1, CallContextFieldTag.CallerAddress, caller.address) \
+        .account_write(caller.address, AccountFieldTag.Nonce, nonce, nonce - 1) \
+        .call_context_read(1, CallContextFieldTag.IsSuccess, is_success) \
+        .call_context_read(1, CallContextFieldTag.IsStatic, is_static) \
+        .call_context_read(1, CallContextFieldTag.RwCounterEndOfReversion, caller_ctx.rw_counter_end_of_reversion) \
+        .call_context_read(1, CallContextFieldTag.IsPersistent, caller_ctx.is_persistent) \
+        .tx_access_list_account_write(1, contract_address, True, is_warm_access, rw_counter_of_reversion=None if caller_ctx.is_persistent else caller_ctx.rw_counter_end_of_reversion - caller_ctx.reversible_write_counter) \
+        .account_write(contract_address, AccountFieldTag.CodeHash, callee_bytecode_hash, 0) \
```

Comment:
And here we should have the new value as `RLC(EMPTY_CODE_HASH)`

Reply:
fixed in [e8d9f05](https://github.com/privacy-scaling-explorations/zkevm-specs/pull/355/commits/e8d9f05bc2a014be74442e84b907966de3af68e1)